opentelemetry/src/sdk/trace/span_processor.rs

```diff
@@ -547,4 +670,108 @@ mod tests {
+        DS: Future<Output = ()> + Send + Sync + 'static,
+    {
+        async fn export(&mut self, batch: Vec<SpanData>) -> ExportResult {
+            println!("Accepting {} spans", batch.len());
+            (self.delay_fn)(self.delay_for).await;
+            println!("Finish exporting, return result from exporter");
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn test_timeout() {
+        // If time_out is true, then we ask exporter to block for 60s and set timeout to 5s.
+        // If time_out is false, then we ask the exporter to block for 5s and set timeout to 60s.
+        // Either way, the test should be finished within 5s.
+        let mut runtime = tokio::runtime::Builder::new()
+            .threaded_scheduler()
+            .enable_all()
+            .build()
+            .unwrap();
+        runtime.block_on(timeout_test_tokio(true));
+        runtime.block_on(timeout_test_tokio(false));
+
+        async_std::task::block_on(timeout_test_std_async(true));
+        async_std::task::block_on(timeout_test_std_async(false));
+    }
+
+    // If the time_out is true, then the result suppose to ended with timeout.
+    // otherwise the exporter should be able to export within time out duration.
+    async fn timeout_test_std_async(time_out: bool) {
+        let mut config = BatchConfig::default();
+        config.max_export_timeout = time::Duration::from_secs(if time_out { 5 } else { 60 });
+        config.scheduled_delay = Duration::from_secs(60 * 60 * 24); // set the tick to 24 hours so we know the span must be exported via force_flush
+        let exporter = BlockingExporter {
+            delay_for: time::Duration::from_secs(if !time_out { 5 } else { 60 }),
+            delay_fn: async_std::task::sleep,
+        };
+        let mut processor = BatchSpanProcessor::new(
+            Box::new(exporter),
+            async_std::task::spawn,
+            async_std::stream::interval,
+            async_std::task::sleep,
+            config,
+        );
+        processor.on_end(new_test_export_span_data());
+        let flush_res = processor.force_flush();
+        if time_out {
+            assert!(flush_res.is_err());
+        } else {
+            assert!(flush_res.is_ok());
+        }
+        let shutdown_res = processor.shutdown();
+        assert!(shutdown_res.is_ok());
+    }
+
+    // If the time_out is true, then the result suppose to ended with timeout.
+    // otherwise the exporter should be able to export within time out duration.
+    async fn timeout_test_tokio(time_out: bool) {
+        let mut config = BatchConfig::default();
+        config.max_export_timeout = time::Duration::from_secs(if time_out { 5 } else { 60 });
+        config.scheduled_delay = Duration::from_secs(60 * 60 * 24); // set the tick to 24 hours so we know the span must be exported via force_flush
+        let exporter = BlockingExporter {
+            delay_for: time::Duration::from_secs(if !time_out { 5 } else { 60 }),
```

Comment:
I think 5 seconds is quite a long time for a test. What kind of delay do we need if we're testing that it does not time out? Would the smallest unit (e.g. 1ms) be enough?